tests/components/rfxtrx/test_light.py

```diff
@@ -147,82 +156,70 @@ async def test_several_lights(hass, rfxtrx):
     )
     await hass.async_block_till_done()
 
-    assert 3 == len(rfxtrx_core.RFX_DEVICES)
-    device_num = 0
-    for id in rfxtrx_core.RFX_DEVICES:
-        entity = rfxtrx_core.RFX_DEVICES[id]
-        assert entity.signal_repetitions == 3
-        if entity.name == "Living":
-            device_num = device_num + 1
-            assert "off" == entity.state
-            assert "<Entity Living: off>" == entity.__str__()
-        elif entity.name == "Bath":
-            device_num = device_num + 1
-            assert "off" == entity.state
-            assert "<Entity Bath: off>" == entity.__str__()
-        elif entity.name == "Test":
-            device_num = device_num + 1
-            assert "off" == entity.state
-            assert "<Entity Test: off>" == entity.__str__()
-
-    assert 3 == device_num
+    assert len(hass.states.async_all()) == 3
 
+    state = hass.states.get("light.test")
+    assert state
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Test"
 
-async def test_discover_light(hass, rfxtrx):
-    """Test with discovery of lights."""
+    state = hass.states.get("light.bath")
+    assert state
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Bath"
+
+    state = hass.states.get("light.living")
+    assert state
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Living"
+
+    assert len(hass.states.async_all()) == 3
+
+
+@pytest.mark.parametrize("repetitions", [1, 3])
+async def test_repetitions(hass, rfxtrx, repetitions):
+    """Test signal repetitions."""
     await async_setup_component(
         hass,
         "light",
-        {"light": {"platform": "rfxtrx", "automatic_add": True, "devices": {}}},
+        {
+            "light": {
+                "platform": "rfxtrx",
+                "signal_repetitions": repetitions,
+                "devices": {"0b1100cd0213c7f230010f71": {"name": "Test"}},
+            }
+        },
     )
     await hass.async_block_till_done()
 
-    event = rfxtrx_core.get_rfx_object("0b11009e00e6116202020070")
-    event.data = bytearray(b"\x0b\x11\x00\x9e\x00\xe6\x11b\x02\x02\x00p")
-
-    await _signal_event(hass, event)
-    entity = rfxtrx_core.RFX_DEVICES["0e61162_2"]
-    assert 1 == len(rfxtrx_core.RFX_DEVICES)
-    assert "<Entity 0b11009e00e6116202020070: on>" == entity.__str__()
-
-    event = rfxtrx_core.get_rfx_object("0b11009e00e6116201010070")
-    event.data = bytearray(b"\x0b\x11\x00\x9e\x00\xe6\x11b\x01\x01\x00p")
+    await hass.services.async_call(
+        "light", "turn_on", {"entity_id": "light.test"}, blocking=True
+    )
+    await hass.async_block_till_done()
 
-    await _signal_event(hass, event)
-    assert 1 == len(rfxtrx_core.RFX_DEVICES)
+    assert rfxtrx.transport.send.call_count == repetitions
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)